workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildContextFilter.cs

```diff
@@ -1,22 +1,91 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Improbable.Gdk.BuildSystem.Configuration;
 using Improbable.Gdk.Core;
 using UnityEditor;
 
-namespace Improbable.Gdk.BuildSystem
+namespace Improbable.Gdk.BuildSystem.Configuration
 {
-    public static class CommandlineParser
+    public struct BuildContextFilter
     {
-        internal static string[] GetWorkerTypesToBuild(CommandLineArgs args)
+        public IEnumerable<string> WantedWorkerTypes;
+        public BuildEnvironment BuildEnvironment;
+        public ScriptingImplementation? ScriptImplementation;
+        public ICollection<BuildTarget> BuildTargetFilter;
+        public iOSSdkVersion? IOSSdkVersion;
+
+        public BuildContextFilter(IEnumerable<string> wantedWorkerTypes, BuildEnvironment buildEnvironment,
+            ScriptingImplementation? scriptImplementation = null, ICollection<BuildTarget> buildTargetFilter = null,
+            iOSSdkVersion? iosSdkVersion = null)
```

Comment:
We can probably drop these optional parameters from the constructor as you can set them with the initializer syntax if you wish:

```csharp
new BuildContextFilter(wantedWorkerTypes, buildEnvironment)
{
	ScriptImplementation = ScriptingImplementation.IL2CPP
}
```

Reply:
I had to add custom setters for scripting/target properties, to make sure ios is always il2cpp